### DISCUSSION TITLE
API Settings Page redirects to default home page in self hosted docker

### DISCUSSION THREAD

**Discussion by omani** [supabase/supabase#37914] [Feature Requests]
I cloned the repo and copied over the docker folder to my project folder. followed the supabase docs for self-hosted supabase with docker.

everything works but I can't access the API settings page. nor any other settings page. it redirects to the home screen (default project home page).

http://localhost:8000/project/default/settings/api does not work.
http://localhost:8000/project/default/settings/ does not work.

**Comment by omani** [comment]
https://supabase.com/docs/guides/self-hosting/docker

**Comment by omani** [comment]
>We have provided you a Client Key to get started. You will soon be able to add as many keys as you like. You can find the anon key in the [API Settings](http://localhost:8000/project/default/settings/api) page.

the "API Settings" link on page http://localhost:8000/project/default/api?page=auth redirects to http://localhost:8000/project/default. the API settings page does not open.

**Comment by Hallidayo** [comment]
Hi @omani - By default the settings pages is disabled for local systems. 

This section here:

> We have provided you a Client Key to get started. You will soon be able to add as many keys as you like. You can find the anon key in the [API Settings](http://localhost:8000/project/default/settings/api) page.

Could you provide a link to this on the docs please?

**Comment by kristianka** [comment]
I can replicate this too. I got the following errors in my console:

![image](https://github.com/user-attachments/assets/bbbee1f1-38f8-4d98-8100-1a0fc58932f7)

I followed this tutorial: https://supabase.com/docs/guides/self-hosting/docker 
Also many other buttons or navigating to http://localhost:8000/project/settings just redirects to the front page.

**Comment by omani** [comment]
>Could you provide a link to this on the docs please?

it is not in the docs. this is in the page (in the studio dashboard) at localhost:8000 under API Guide or API Docs or something like that in the navbar. can't remember anymore (dont have studio opened right now).

**Comment by saltcod** [comment]
Thanks for the reports all. 
Curious where you see those links? We need to check if self hosted env and not output them, as we don't expose settings in this way on self hosted. We use a config.toml file for this in that environment instead.

**Comment by abhinavdas** [comment]
![Screenshot 2024-09-27 at 8 10 52 PM](https://github.com/user-attachments/assets/5e4eaaf5-114e-4f8b-9033-d8b864bd9979)
I am having the same redirect issue but the reason I want to reach the settings is due to this issue of public schema not being exposed. I am not sure what to do next.

**Comment by omani** [comment]
> Curious where you see those links?

like I said it's right there on the web UI. click on the `API Guide` link in the navbar.

**Comment by olb-openminds** [comment]
A temporary solution if you are using docker and docker desktop.
In containers 
supabase 
 -> supabase-kong 
open files

in path /home/kong/kong.yml 
![image](https://github.com/user-attachments/assets/0c8d68a1-01d5-4bd1-83a2-fe051c0f701e)

you should be able to see 
consumers: 
 username: anon 
 keyauth_credentials:
      - key: XXXX

**Comment by Hallidayo** [comment]
Hi all, We’re going to close this one due to inactivity. Please feel free to reopen if you’re still having issues here or open a new issue.

**Comment by Cloudore** [comment]
Sorry, I see this was closed just yesterday but I am currently experiencing this same issue

I have made a video of the issue

[https://imgur.com/a/0RJFOG1](url)

@Hallidayo Could you reopen the issue? I have tried everything but can't fix it myself

`realtime-dev.supabase-realtime  | 05:45:12.185 request_id=GCq9oqUlmK9qSPoAACQB [info] Sent 403 in 243µs
supabase-imgproxy               | INFO    [2025-03-08T05:45:12Z] Started /health  request_id=GRWRDbABotxBzFTWQ07KD method=GET client_ip=127.0.0.1
supabase-imgproxy               | INFO    [2025-03-08T05:45:12Z] Completed in 62.019µs /health  request_id=GRWRDbABotxBzFTWQ07KD method=GET status=200 client_ip=127.0.0.1
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "GET /api/platform/projects/default HTTP/1.1" 304 0 "http://192.168.187.207:8001/project/default/api" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "GET /api/platform/integrations/github/connections?organization_id=1 HTTP/1.1" 304 0 "http://192.168.187.207:8001/project/default/api" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "GET /api/platform/organizations/default-org-slug/billing/subscription HTTP/1.1" 304 0 "http://192.168.187.207:8001/project/default/api" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "GET /api/platform/projects/default/settings HTTP/1.1" 304 0 "http://192.168.187.207:8001/project/default/api" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "GET /api/platform/projects/default/run-lints HTTP/1.1" 304 0 "http://192.168.187.207:8001/project/default/api" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-kong                   | 172.18.0.10 - - [08/Mar/2025:05:45:12 +0000] "GET /rest/v1/ HTTP/1.1" 200 2206 "-" "node"
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "GET /api/platform/projects/default/api/rest HTTP/1.1" 304 0 "http://192.168.187.207:8001/project/default/api" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-kong                   | 2025/03/08 05:45:12 [warn] 1110#0: *3609 a client request body is buffered to a temporary file /usr/local/kong/client_body_temp/0000000011, client: 192.168.187.160, server: kong, request: "POST /api/platform/pg-meta/default/query?key=entity-definitions HTTP/1.1", host: "192.168.187.207:8001", referrer: "http://192.168.187.207:8001/project/default/api"
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "GET /api/platform/projects/default/config/postgrest HTTP/1.1" 304 0 "http://192.168.187.207:8001/project/default/api" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-kong                   | 172.18.0.10 - - [08/Mar/2025:05:45:12 +0000] "GET /rest/v1/ HTTP/1.1" 200 2206 "-" "node"
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "GET /api/platform/projects HTTP/1.1" 304 0 "http://192.168.187.207:8001/project/default/api" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "GET /api/platform/projects/default/api/rest HTTP/1.1" 304 0 "http://192.168.187.207:8001/project/default/api" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "GET /api/platform/projects/default/config/postgrest HTTP/1.1" 304 0 "http://192.168.187.207:8001/project/default" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "GET /api/platform/integrations/github/connections?organization_id=1 HTTP/1.1" 304 0 "http://192.168.187.207:8001/project/default" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-kong                   | 172.18.0.10 - - [08/Mar/2025:05:45:12 +0000] "GET /auth/v1/settings HTTP/1.1" 200 521 "-" "node"
supabase-auth                   | {"level":"info","msg":"Request received external host in X-Forwarded-Host or Host headers, but the values have not been added to GOTRUE_MAILER_EXTERNAL_HOSTS and will not be used. To suppress this message add the host, or sanitize the headers before the request reaches Auth.","time":"2025-03-08T05:45:12Z","x_forwarded_host":"kong","x_forwarded_proto":"http"}
supabase-auth                   | {"component":"api","duration":782018,"level":"info","method":"GET","msg":"request completed","path":"/settings","referer":"http://localhost:3000","remote_addr":"172.18.0.10","request_id":"7c42949f-31aa-4791-a670-c2b87a39ab9b","status":200,"time":"2025-03-08T05:45:12Z"}
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "GET /api/platform/auth/default/config HTTP/1.1" 200 1419 "http://192.168.187.207:8001/project/default" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "GET /api/platform/props/project/default/jwt-secret-update-status HTTP/1.1" 404 2691 "http://192.168.187.207:8001/project/default" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "POST /api/platform/pg-meta/default/query?key=schemas HTTP/1.1" 200 786 "http://192.168.187.207:8001/project/default" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "GET /api/platform/pg-meta/default/extensions HTTP/1.1" 304 0 "http://192.168.187.207:8001/project/default" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-analytics              | 
supabase-analytics              | 05:45:12.503 [info] All logs logged!
supabase-analytics              | 
supabase-analytics              | 05:45:12.503 [info] Logs last second!
supabase-kong                   | 192.168.187.160 - - [08/Mar/2025:05:45:12 +0000] "POST /api/platform/pg-meta/default/query?key=entity-definitions HTTP/1.1" 200 29 "http://192.168.187.207:8001/project/default/api" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.3 Safari/605.1.15"
supabase-analytics              | `

**Comment by Hallidayo** [comment]
Hi @Cloudore , I've just reopened and will take a look.

**Comment by Cloudore** [comment]
> Hi [@Cloudore](https://github.com/Cloudore) , I've just reopened and will take a look.

Thank you, I tried on a fresh computer with all default credentials and no cigar, same issue

I have summarized the logs with chatgpt

`Key Points in the Log:

	•	HTTP Requests:

	•	For example, the first entry shows a GET /auth/v1/settings request that returned a 200 status with 521 bytes.

	•	Another line shows a POST /api/platform/pg-meta/default/query?key=schemas request that returned 200—

indicating successful processing.

	•	Status Codes:

	•	200: Indicates a successful request.

	•	404: The entry for GET /api/platform/props/project/default/jwt-secret-update-status returned 404, suggesting that this endpoint was not found.

	•	499: The request for /api/platform/pg-meta/default/extensions returned a 499, which typically means the client closed the connection before the server could send a response.`

**Comment by whogben** [comment]
Hi all, I am experiencing this issue as well.

I have found two links to settings, one on the API docs, and the other on the Storage page. I was following a tutorial and it told me to visit the settings - settings loads for a moment and then redirects to home page.

@Hallidayo you said "Hi @omani - By default the settings pages is disabled for local systems." - can you say how to change that default to enable it?

Here's a video of the issue - showing two different links to settings in UI that I'd like to access (API settings, Project settings):

https://github.com/user-attachments/assets/917abd3f-70d4-451e-8fb9-c323ed3241e9

Here's screenshots of the settings page that loads and is clearly present for a moment before the redirect:
<img width="810" alt="Image" src="https://github.com/user-attachments/assets/e9871408-fb9b-44ce-b7b9-0b07d1b1584c" />
<img width="796" alt="Image" src="https://github.com/user-attachments/assets/80bd7bb0-71fb-4bc4-9a29-c930932c6935" />

Is there some configuration I can do to stop the redirect - or is there a documentation page on what to do when a tutorial references supabase settings that are normally on the page, eg, how to adapt them for use via alternative methods?

**Comment by charislam** [comment]
Hi @whogben, those settings pages are disabled on self-hosted because they use Supabase's cloud platform API -- even if the redirect was removed, the pages wouldn't be useful because your self-hosted projects aren't managed by our management API and the endpoints to update settings wouldn't work!

In general, config settings can be changed using the configuration variables defined in our self-hosted reference guides: https://supabase.com/docs/guides/self-hosting

We can definitely do a better job of documenting these and making the information obvious in self-hosted dashboard, it is on our radar!

**Comment by whogben** [comment]
Ah ok, thanks for clarifying, yeah I think if there weren't links to the settings still present in self hosted I would have gotten it. Sorry it took me so long.

**Comment by Nikto655** [comment]
This is still an issue in self-hosted version. Cant access the setting page at all.

**Comment by aminebeh** [comment]
HI @charislam, thanks for your reply.

I've started using supabase a few weeks ago and the product is amazing, it's great and thanks for open-sourcing it.

I think this kind of feature (api, database and other settings page) can really improve the experience on self-hosted version (and might help for maintaining docs between cloud and self-hosted versions).

I get the reason on the technical side (dependency on cloud platform API), but is there a chance that you add supports for these pages on self-hosted plans later ?

Thanks in advance for you works and consideration !

**Comment by Hallidayo** [comment]
Hi, all I'm going to move this over as a discussion/enhancement. Thanks for all your input on this one.

**Comment by maxmckenzie** [comment]
So i've noticed that once this had been made into a discussion, It kinda died. BUt i've got a reason to respond to this

So i'm currently using Supbase functions with Hono. I was trying to follow the guide to use the new JWT token stuff. But I need to rewrite the auth middelware I use in the functions. Which i tried to do.

I'm trying to find the service key again, but i cant, and now after updating supabase, when you do `supabase status` the service key is gone, it only shows the new publishable and secret key? and i cant find anywhere in localhost to retrieve the keys. Its in the settings page, but of course its local so there is no settings page.

Its pretty fustrating and they are discontinuing the old way of doing API keys anon and service key in late 2026. So.... thers a time limit on this as well. Any help would be much appreciated.
